geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/data/KafkaCacheLoader.scala

```diff
@@ -114,12 +124,13 @@ object KafkaCacheLoader {
     }
 
     override protected [KafkaCacheLoader] def consume(record: ConsumerRecord[Array[Byte], Array[Byte]]): Unit = {
-      val message = serializer.deserialize(record.key, record.value)
+      val message = serializer.deserialize(record.key(), record.value())
+      val timestamp = try { record.timestamp() } catch { case e: NoSuchMethodError => System.currentTimeMillis() }
       logger.trace(s"Consumed message [$topic:${record.partition}:${record.offset}] $message")
       message match {
-        case m: Change => fireEvent(m); cache.put(m.feature)
-        case m: Delete => fireEvent(m); cache.remove(m.id)
-        case m: Clear  => fireEvent(m); cache.clear()
+        case m: Change => fireEvent(m, timestamp); cache.put(m.feature)
```

Comment:
Actually, since having listeners subscribed is fairly rare.... can you move the calculation of the timestamp inside the guarded part of the fireEvent?

Reply:
While we are on this thread...  I wonder if the fireEvent's should be var's which are no-ops until they are reset.  This'd avoid the boolean checks.

Reply:
the timestamp should just be a field accessor, unless you're using an older version of kafka (0.9 or older i think). To move it to the guarded part, we'd either have to pass the whole record through the api, or pass a lambda. But i'd rather not degrade the regular path for slightly better support of old kafka versions. what do you think?

Reply:
i kind of wonder if a no-op lambda is actually faster than a boolean check - i've been leaning towards the latter, as it seems the type of thing that the processor will optimize (i.e. when 100% of the time you go through one branch it will start skipping it aggressively and backtrack if needed).